vault/plugin_reload.go

```diff
@@ -0,0 +1,103 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+// reloadPluginMounts reloads provided mounts, regardless of
+// plugin name, as long as the backend type is of plugin.
+func (c *Core) reloadMatchingPluginMounts(mounts []string) error {
+	c.mountsLock.Lock()
+	defer c.mountsLock.Unlock()
+
+	for _, mount := range mounts {
+		entry := c.router.MatchingMountEntry(mount)
+		if entry == nil {
+			return fmt.Errorf("cannot fetch mount entry on %s", mount)
+		}
+		err := c.reloadPluginCommon(entry)
+		if err != nil {
+			return err
+		}
+		c.logger.Info("core: successfully reloaded plugin", "plugin", entry.Config.PluginName, "path", entry.Path)
+	}
+	return nil
+}
+
+// reloadPlugin reloads all mounted backends that are of
+// plugin pluginName (name of the plugin as registered in
+// the plugin catalog).
+func (c *Core) reloadMatchingPlugin(pluginName string) error {
+	c.mountsLock.Lock()
+	defer c.mountsLock.Unlock()
+
+	// Filter mount entries that only matches the plugin name
+	for _, entry := range c.mounts.Entries {
```

Comment:
This function likely needs to iterate through c.auth.Entries too, in case the plugin name they give is an auth plugin.